---
title: "TCox: correlation-based regularization applied to colorectal cancer"
author: "Carolina Peixoto, Marta B. Lopes, Marta Martins, Lu√≠s Costa and Susana Vinga"
date: "September, 2020"
output:
  html_document: 
    number_sections: yes
    toc: yes
params:
  seed: !r 3390
  train: !r 0.7
  nfolds: !r 10
editor_options:
  chunk_output_type: inline
---


# Install packages and functions
```{r}
devtools::install_github("sysbiomed/glmSparseNet", ref = "prognostic.index")
```
## Required libraries
```{r message = FALSE}
library(SummarizedExperiment)
library(dplyr)
library(DT)
library(tidyverse)
library(futile.logger)
library(loose.rock)
library(survival)
library(GGally)
library(ggplot2)
library(survminer)
library(glmnet)
library(glmSparseNet)
library(gplots)
library(readr)
library(maftools)
library(RTCGAToolbox)
library(edgeR)
#install.packages("biospear")
library(biospear)
```


## Functions
```{r message = FALSE}
twiner_glmnet <- function (x,y,alpha, w) {
                      cv.glmnet(as.matrix(x),as.matrix(y), 
                               family = 'cox',foldid = my_foldid, penalty.factor = w, 
                               alpha = alpha,
                               nlambda = 100,
                               #lambda.min.ratio = 10^(-2),
                               nfolds= 10)
}

twiner_coefs <- function (cv.fit){
  coef(cv.fit, s = 'lambda.min')[,1] %>% { .[. != 0]}
}


twiner_genes <- function (cv.fit) {
  b <- which(cv.fit$glmnet.fit$beta[,which(cv.fit$cvm == min(cv.fit$cvm))] != 0)
  return (names(b))
}

train <- function (coefs) {
  separate2GroupsCox(as.vector(coefs), 
                               xdata.train[, names(coefs)], 
                               ydata.train, 
                               plot.title = 'Train Set TCox', legend.outside = FALSE) 
}

test <- function (coefs) {
  separate2GroupsCox(as.vector(coefs), 
                               xdata.test[, names(coefs)], 
                               ydata.test, 
                               plot.title = 'Test Set TCox', legend.outside = FALSE) 
}

all <- function (coefs) {
  separate2GroupsCox(as.vector(coefs), 
                               xdata[, names(coefs)], 
                               ydata, 
                               plot.title = 'all Set TCox', legend.outside = FALSE) 
}

```

# Download and prepare dataframes from TCGA - RTCGAtoolbox package
## Colorectal samples RNASeq data

Download COAD dataset

```{r,warning=FALSE,message=FALSE}
coadData <- getFirehoseData(dataset="COAD", runDate="20150402", 
    forceDownload=TRUE, clinical=TRUE, RNASeq2GeneNorm=TRUE) 

rnaseq1 <- coadData@RNASeq2GeneNorm[[2]]
rnaseq1 <- rnaseq1@DataMatrix
rnaseq1 <- t(rnaseq1)
```

```{r,warning=FALSE,message=FALSE}
rnaseq1 <- as.data.frame(rnaseq1)


rnaseq1$names <- rownames(rnaseq1)


dim(rnaseq1) #328

#keep only tumor samples
tumor <- rnaseq1 %>%
  filter(str_detect(names, "01A"))

dim(tumor) #287
tumor$names <- str_sub(tumor$names, 1, str_length(tumor$names)-16) #put names in the right format
#remove duplicated rows
tumor <- tumor[!duplicated(tumor$names), ]
rownames(tumor) <- tumor$names
tumor <- tumor[,-20502]

normal <- rnaseq1 %>%
  filter(!str_detect(names, "01A"))

dim(normal) #287
normal$names <- str_sub(normal$names, 1, str_length(normal$names)-16) #put names in the right format
normal <- normal[!duplicated(normal$names), ]
rownames(normal) <- normal$names
normal <- normal[,-20502]




```

Download READ dataset

```{r, warning=FALSE,message=FALSE}
readData <- getFirehoseData(dataset="READ", runDate="20150402", 
    forceDownload=TRUE, clinical=TRUE, RNASeq2GeneNorm=TRUE) 

rnaseq2 <- readData@RNASeq2GeneNorm[[2]]
rnaseq2 <- rnaseq2@DataMatrix
rnaseq2 <- t(rnaseq2)
```

```{r,warning=FALSE,message=FALSE}
rnaseq2 <- as.data.frame(rnaseq2)


rnaseq2$names <- rownames(rnaseq2)


dim(rnaseq2) #328

#keep only tumor samples
tumorread <- rnaseq2 %>%
  filter(str_detect(names, "01A"))

dim(tumorread) #287
tumorread$names <- str_sub(tumorread$names, 1, str_length(tumorread$names)-16) #put names in the right format
#remove duplicated rows
tumorread <- tumorread[!duplicated(tumorread$names), ]
rownames(tumorread) <- tumorread$names
tumorread <- tumorread[,-20502]

normalread <- rnaseq2 %>%
  filter(!str_detect(names, "01A"))

dim(normalread) #287
normalread$names <- str_sub(normalread$names, 1, str_length(normalread$names)-16) #put names in the right format
normalread <- normalread[!duplicated(normalread$names), ]
rownames(normalread) <- normalread$names
normalread <- normalread[,-20502]

```

Colorectal dataset

```{r,warning=FALSE,message=FALSE}
dim(tumor)
dim(normal)
dim(tumorread)
dim(normalread)

tumorf <- rbind(tumor,tumorread)
dim(tumorf)

normalf <- rbind(normal, normalread)
dim(normalf)

# write.table(tumorf, "crcRNAseq.txt")
# write.table(normalf, "normalRNAseq.txt")

#write.table(datasurvf, "datasurv.txt")
```

## Clinical data

COAD

```{r,warning=FALSE,message=FALSE}
clinic <- coadData@clinical

clinic.data <- clinic[,3:5]
clinic.data$days_to_death[is.na(clinic.data$days_to_death)] <- 0

for (i in 1:nrow(clinic.data)){
  
  if (clinic.data$days_to_death[i] != 0){
    
    clinic.data$days_to_last_followup[i] <- 0
    
  }
  
datasurv <- apply(as.matrix(clinic.data),2,as.numeric)
datasurv <- as.data.frame(datasurv)
datasurv$time <- datasurv$days_to_death + datasurv$days_to_last_followup

datasurv <- datasurv[,-c(2,3)]
rownames(datasurv) <- rownames(clinic.data)
datasurv <- datasurv[datasurv$time > 0, ] 
datasurv <- na.omit(datasurv)
colnames(datasurv) <- c("status","time")

datasurv$names <- rownames(datasurv)
datasurv$names <- toupper(datasurv$names)  
datasurv$names <- chartr('.', '-', datasurv$names)

rownames(datasurv) <- datasurv$names
datasurv <- datasurv[,-3]
}
  
  
```

READ

```{r,warning=FALSE,message=FALSE}
clinicread <- readData@clinical

clinicread.data <- clinicread[,3:5]
clinicread.data$days_to_death[is.na(clinicread.data$days_to_death)] <- 0

for (i in 1:nrow(clinicread.data)){
  
  if (clinicread.data$days_to_death[i] != 0){
    
    clinicread.data$days_to_last_followup[i] <- 0
    
  }
  
datasurvread <- apply(as.matrix(clinicread.data),2,as.numeric)
datasurvread <- as.data.frame(datasurvread)
datasurvread$time <- datasurvread$days_to_death + datasurvread$days_to_last_followup

datasurvread <- datasurvread[,-c(2,3)]
rownames(datasurvread) <- rownames(clinicread.data)
datasurvread <- datasurvread[datasurvread$time > 0, ] 
datasurvread <- na.omit(datasurvread)
colnames(datasurvread) <- c("status","time")

datasurvread$names <- rownames(datasurvread)
datasurvread$names <- toupper(datasurvread$names)  
datasurvread$names <- chartr('.', '-', datasurvread$names)

rownames(datasurvread) <- datasurvread$names
datasurvread <- datasurvread[,-3]
}
  
  
```

Colorectal 

```{r,warning=FALSE,message=FALSE}
dim(datasurv)
dim(datasurvread)

datasurvf <- rbind(datasurv,datasurvread)
dim(datasurvf)

clinicalf <- rbind(clinic,clinicread)
```


# TCox 

## using both normal and tumor data from TCGA

- removing variables with sd=0

```{r message = FALSE}
xcoad_sd <- tumorf [,sapply(seq(ncol(tumorf)), function(ix) {sd(tumorf[,ix])}) != 0] # 328 20025
xnormal_sd <- normalf[,sapply(seq(ncol(normalf)), function(ix) {sd(normalf[,ix])}) != 0] #345 48710

xcoad_less <- xcoad_sd[,which(colnames(xcoad_sd) %in% colnames(xnormal_sd))]
xnormal_less <- xnormal_sd[,which(colnames(xnormal_sd) %in% colnames(xcoad_sd))]

# normalizing data
xcoad_norm <- scale(log2(xcoad_less+1)) 
xnorm_norm <- scale(log2(xnormal_less+1)) 
```

- weight vector that penalizes genes with smaller distances between tumor and normal correlation matrices

```{r message = FALSE}
## correlation matrices

library("propagate")
library("lsa")

options(fftempdir = "~/")

xtumoral_cor <- bigcor(xcoad_norm, y = NULL, fun = "cor", size = 2000, verbose=FALSE)
xtumoral_cor <- as.data.frame(as.ffdf(xtumoral_cor))

xnormal_cor <- bigcor(xnorm_norm, y = NULL, fun = "cor", size = 2000, verbose=FALSE)
xnormal_cor <- as.data.frame(as.ffdf(xnormal_cor))

# angular distance
ang_weight <- vector()

for (i in 1:dim(xcoad_norm)[2]){
ang_weight[i] <- acos(cosine(xtumoral_cor[,i],xnormal_cor[,i]))/pi
}

```

```{r,warning=FALSE,message=FALSE}
##normalized weights (between 0 and 1)

tumornormal_weights <- ang_weight / max(ang_weight)
hist(ang_weight)
hist(tumornormal_weights,main="normal weight")

#1-w
tumornormal_weights1 <- 1 - tumornormal_weights
hist(tumornormal_weights1,main="1 - normal weight")

#1-w^3
tumornormal_weights2 <- 1 - (tumornormal_weights^3)
hist(tumornormal_weights2,main="1 - normal weight^3")

#(1-w)^3
tumornormal_weights3 <- (1 - tumornormal_weights)^3
hist(tumornormal_weights3,main="(1 - normal weight)^3")

#exp((1-w)^3)
tumornormal_weights4 <- exp((1 - tumornormal_weights)^3)
hist(tumornormal_weights4,main="exp((1 - normal weight)^3)")

#exp(-w^3)
tumornormal_weights5 <- exp(- tumornormal_weights^3)
hist(tumornormal_weights5,main="exp(- normal weight^3)")

#1/w
tumornormal_weights6 <- 1 / tumornormal_weights
hist(tumornormal_weights6,main="1 / normal weight")
```

## Analyse different weight vectores:

```{r,warning=FALSE,message=FALSE}
xdata.raw <- as.data.frame(xcoad_less) #255 19525

ydata.raw <- as.data.frame(datasurvf)
ydata.raw$rownames <- rownames(ydata.raw)
ydata.raw$rownames <- chartr('.', '-', ydata.raw$rownames)
rownames(ydata.raw) <- ydata.raw$rownames
ydata <- ydata.raw[rownames(ydata.raw) %in% 
                              rownames(xdata.raw),]

  
xdata.raw <- xdata.raw[rownames(xdata.raw) %in% 
                         rownames(ydata),]
xdata <- xdata.raw %>% 
  { (apply(., 2, sd) != 0) } %>% 
  { xdata.raw[, .] } %>% 
  scale

xdata <- as.matrix(xdata) #293 17910

ydata <- ydata[,1:2]
dim(xdata)
```

Test / Training sets
```{r,warning=FALSE,message=FALSE}
set.seed(params$seed)
ixs <- loose.rock::balanced.train.and.test(which(as.logical(ydata$status)), which(as.logical(!ydata$status)), train.perc = params$train)
xdata.test <- xdata[ixs$test,]
ydata.test <- ydata[ixs$test,]
#
xdata.train <- xdata[ixs$train,]
ydata.train <- ydata[ixs$train,]

xdata.train.digest <- loose.rock::digest.cache(xdata.train)

flog.info('Size of sets: (size/events)\n * Training set: % 4d/% 4d\n * Test set:     % 4d/% 4d', 
          nrow(xdata.train), 
          sum(ydata.train$status), 
          nrow(xdata.test),
          sum(ydata.test$status))

 # * Training set:  204/  50
 # * Test set:       89/  22

#Foldid
my_foldid <- sample(1:10,size=dim(ydata.train)[1],replace=TRUE)


#Remove columns with standard deviation of zero
xdata.train <- xdata.train[, ! apply(xdata.train , 2 , function(x) sd(x, na.rm = TRUE)==0 ) ]

xdata.test <- xdata.test[, ! apply(xdata.test , 2 , function(x) sd(x, na.rm = TRUE)==0 ) ]

dim(xdata.test)
dim(xdata.train)

xdata.train <- xdata.train[,colnames(xdata.train) %in% 
                         colnames(xdata.test)]
xdata.test <- xdata.test[,colnames(xdata.test) %in% 
                         colnames(xdata.train)]

dim(xdata.test)
dim(xdata.train)

xdata.train[1:3,1:3]
```



### 1/w

#### 0.3
```{r,warning=FALSE,message=FALSE}
#glmnet
cv.fit_twii3 <- twiner_glmnet(xdata.train,ydata.train,0.3, tumornormal_weights6)


#genes
coefs.v_twii3 <- twiner_coefs(cv.fit_twii3)

fit.selected_twii3 <- twiner_genes(cv.fit_twii3)
length(fit.selected_twii3)
fit.selected_twii3

#train
train_twii03 <- train (coefs.v_twii3)
train_twii03$pvalue ## these values are the ones in Table 1
#test
test_twii03 <- test (coefs.v_twii3)
test_twii03$pvalue
#all
all_twii03 <- all (coefs.v_twii3)

```
#### 0.2
```{r,warning=FALSE,message=FALSE}
#glmnet
cv.fit_twii2 <- twiner_glmnet(xdata.train,ydata.train,0.2, tumornormal_weights6)


#genes
coefs.v_twii2 <- twiner_coefs(cv.fit_twii2)

fit.selected_twii2 <- twiner_genes(cv.fit_twii2)
length(fit.selected_twii2)
fit.selected_twii2

#train
train_twii02 <- train (coefs.v_twii2)
train_twii02$pvalue
#test
test_twii02 <- test (coefs.v_twii2)
test_twii02$pvalue
#all
all_twii02 <- all (coefs.v_twii2)

```

#### 0.1
```{r,warning=FALSE,message=FALSE}
#glmnet
cv.fit_twii1 <- twiner_glmnet(xdata.train,ydata.train,0.1, tumornormal_weights6)

#genes
coefs.v_twii1 <- twiner_coefs(cv.fit_twii1)

fit.selected_twii1 <- twiner_genes(cv.fit_twii1)
length(fit.selected_twii1)
fit.selected_twii1

#train
train_twii01 <- train (coefs.v_twii1)
train_twii01$pvalue
#test
test_twii01 <- test (coefs.v_twii1)
test_twii01$pvalue
#all
all_twii01 <- all (coefs.v_twii1)

```

#### 0.05
```{r,warning=FALSE,message=FALSE}
#glmnet
cv.fit_twii05 <- twiner_glmnet(xdata.train,ydata.train,0.05, tumornormal_weights6)

#genes
coefs.v_twii05 <- twiner_coefs(cv.fit_twii05)

fit.selected_twii05 <- twiner_genes(cv.fit_twii05)
length(fit.selected_twii05)


#train
train_twii05 <- train (coefs.v_twii05)

#test
test_twii05 <- test (coefs.v_twii05)

#all
all_twii05 <- all (coefs.v_twii05)

```


#### plot

```{r,warning=FALSE,message=FALSE}
alpha_points<-c(0.3, 0.2,0.1,0.05)

pvalues_twii_values_train<-c(train_twii03$pvalue,train_twii02$pvalue,train_twii01$pvalue,train_twii05$pvalue)

pvalues_twii_values_test<-c(test_twii03$pvalue,test_twii02$pvalue,test_twii01$pvalue,test_twii05$pvalue)

pvalues_twii_values_all<-c(all_twii03$pvalue,all_twii02$pvalue,all_twii01$pvalue,all_twii05$pvalue)



pvalues_twii_values<-c(train_twii03$pvalue,train_twii02$pvalue,train_twii01$pvalue,train_twii05$pvalue,test_twii03$pvalue,test_twii02$pvalue,test_twii01$pvalue,test_twii05$pvalue,all_twii03$pvalue,all_twii02$pvalue,all_twii01$pvalue,all_twii05$pvalue)
                    


pvalues_twii_method<-c(rep("pvalues TCox (1/w) train set",4),rep("pvalues TCox (1/w) test set",4),rep("pvalues TCox (1/w) all set",4))




pvalues_twii<-data.frame(alpha_points,pvalues_twii_values,pvalues_twii_method)
p_pvalues_twii<-ggplot(data = pvalues_twii, aes(x = alpha_points, y = pvalues_twii_values,color= pvalues_twii_method)) + geom_line()+geom_point()+theme_minimal()
p_pvalues_twii+ ylim(0, max(pvalues_twii_values))
```


### exp(-w^3)

#### 0.3
```{r,warning=FALSE,message=FALSE}
#glmnet
cv.fit_twexp23 <- twiner_glmnet(xdata.train,ydata.train,0.3, tumornormal_weights5)


#genes
coefs.v_twexp23 <- twiner_coefs(cv.fit_twexp23)

fit.selected_twexp23 <- twiner_genes(cv.fit_twexp23)
length(fit.selected_twexp23)


#train
train_twexp203 <- train (coefs.v_twexp23)
train_twexp203$pvalue ## these values are the ones in Table 1
#test
test_twexp203 <- test (coefs.v_twexp23)
test_twexp203$pvalue
#all
all_twexp203 <- all (coefs.v_twexp23)

```


#### 0.2
```{r,warning=FALSE,message=FALSE}
#glmnet
cv.fit_twexp22 <- twiner_glmnet(xdata.train,ydata.train,0.2, tumornormal_weights5)


#genes
coefs.v_twexp22 <- twiner_coefs(cv.fit_twexp22)

fit.selected_twexp22 <- twiner_genes(cv.fit_twexp22)
length(fit.selected_twexp22)

#train
train_twexp202 <- train (coefs.v_twexp22)

#test
test_twexp202 <- test (coefs.v_twexp22)

#all
all_twexp202 <- all (coefs.v_twexp22)

```

#### 0.1
```{r,warning=FALSE,message=FALSE}
#glmnet
cv.fit_twexp21 <- twiner_glmnet(xdata.train,ydata.train,0.1, tumornormal_weights5)

#genes
coefs.v_twexp21 <- twiner_coefs(cv.fit_twexp21)

fit.selected_twexp21 <- twiner_genes(cv.fit_twexp21)
length(fit.selected_twexp21)

#train
train_twexp201 <- train (coefs.v_twexp21)

#test
test_twexp201 <- test (coefs.v_twexp21)

#all
all_twexp201 <- all (coefs.v_twexp21)

```

#### 0.05
```{r,warning=FALSE,message=FALSE}
#glmnet
cv.fit_twexp205 <- twiner_glmnet(xdata.train,ydata.train,0.05, tumornormal_weights5)

#genes
coefs.v_twexp205 <- twiner_coefs(cv.fit_twexp205)

fit.selected_twexp205 <- twiner_genes(cv.fit_twexp205)
length(fit.selected_twexp205)

#train
train_twexp205 <- train (coefs.v_twexp205)

#test
test_twexp205 <- test (coefs.v_twexp205)

#all
all_twexp205 <- all (coefs.v_twexp205)

```



#### plot

```{r,warning=FALSE,message=FALSE}
alpha_points<-c(0.3, 0.2,0.1,0.05)

pvalues_twexp2_values_train<-c(train_twexp203$pvalue,train_twexp202$pvalue,train_twexp201$pvalue,train_twexp205$pvalue)

pvalues_twexp2_values_test<-c(test_twexp203$pvalue,test_twexp202$pvalue,test_twexp201$pvalue,test_twexp205$pvalue)

pvalues_twexp2_values_all<-c(all_twexp203$pvalue,all_twexp202$pvalue,all_twexp201$pvalue,all_twexp205$pvalue)



pvalues_twexp2_values<-c(train_twexp203$pvalue,train_twexp202$pvalue,train_twexp201$pvalue,train_twexp205$pvalue,test_twexp203$pvalue,test_twexp202$pvalue,test_twexp201$pvalue,test_twexp205$pvalue,all_twexp203$pvalue,all_twexp202$pvalue,all_twexp201$pvalue,all_twexp205$pvalue)
                    


pvalues_twexp2_method<-c(rep("pvalues TCox exp(-w^3) train set",4),rep("pvalues TCox exp(-w^3)  test set",4),rep("pvalues TCox exp(-w^3)  all set",4))




pvalues_twexp2<-data.frame(alpha_points,pvalues_twexp2_values,pvalues_twexp2_method)
p_pvalues_twexp2<-ggplot(data = pvalues_twexp2, aes(x = alpha_points, y = pvalues_twexp2_values,color= pvalues_twexp2_method)) + geom_line()+geom_point()+theme_minimal()
p_pvalues_twexp2+ ylim(0, max(pvalues_twexp2_values))
```



### 1-W

#### 0.3
```{r,warning=FALSE,message=FALSE}
#glmnet
cv.fit_tw1i3 <- twiner_glmnet(xdata.train,ydata.train,0.3, tumornormal_weights1)


#genes
coefs.v_tw1i3 <- twiner_coefs(cv.fit_tw1i3)

fit.selected_tw1i3 <- twiner_genes(cv.fit_tw1i3)
length(fit.selected_tw1i3)


#train
train_tw1i03 <- train (coefs.v_tw1i3)

#test
test_tw1i03 <- test (coefs.v_tw1i3)

#all
all_tw1i03 <- all (coefs.v_tw1i3)

```

#### 0.2
```{r,warning=FALSE,message=FALSE}
#glmnet
cv.fit_tw1i2 <- twiner_glmnet(xdata.train,ydata.train,0.2, tumornormal_weights1)


#genes
coefs.v_tw1i2 <- twiner_coefs(cv.fit_tw1i2)

fit.selected_tw1i2 <- twiner_genes(cv.fit_tw1i2)
length(fit.selected_tw1i2)


#train
train_tw1i02 <- train (coefs.v_tw1i2)

#test
test_tw1i02 <- test (coefs.v_tw1i2)

#all
all_tw1i02 <- all (coefs.v_tw1i2)

```

#### 0.1
```{r,warning=FALSE,message=FALSE}
#glmnet
cv.fit_tw1i1 <- twiner_glmnet(xdata.train,ydata.train,0.1, tumornormal_weights1)

#genes
coefs.v_tw1i1 <- twiner_coefs(cv.fit_tw1i1)

fit.selected_tw1i1 <- twiner_genes(cv.fit_tw1i1)
length(fit.selected_tw1i1)

#train
train_tw1i01 <- train (coefs.v_tw1i1)

#test
test_tw1i01 <- test (coefs.v_tw1i1)

#all
all_tw1i01 <- all (coefs.v_tw1i1)

```

#### 0.05
```{r,warning=FALSE,message=FALSE}
#glmnet
cv.fit_tw1i05 <- twiner_glmnet(xdata.train,ydata.train,0.05, tumornormal_weights1)

#genes
coefs.v_tw1i05 <- twiner_coefs(cv.fit_tw1i05)

fit.selected_tw1i05 <- twiner_genes(cv.fit_tw1i05)
length(fit.selected_tw1i05)


#train
train_tw1i05 <- train (coefs.v_tw1i05)

#test
test_tw1i05 <- test (coefs.v_tw1i05)

#all
all_tw1i05 <- all (coefs.v_tw1i05)

```


#### plot

```{r,warning=FALSE,message=FALSE}
alpha_points<-c(0.3,0.2,0.1,0.05)

pvalues_tw1i_values_train<-c(train_tw1i03$pvalue,train_tw1i02$pvalue,train_tw1i01$pvalue,train_tw1i05$pvalue)

pvalues_tw1i_values_test<-c(test_tw1i03$pvalue,test_tw1i02$pvalue,test_tw1i01$pvalue,test_tw1i05$pvalue)

pvalues_tw1i_values_all<-c(all_tw1i03$pvalue,all_tw1i02$pvalue,all_tw1i01$pvalue,all_tw1i05$pvalue)



pvalues_tw1i_values<-c(train_tw1i03$pvalue,train_tw1i02$pvalue,train_tw1i01$pvalue,train_tw1i05$pvalue,test_tw1i03$pvalue,test_tw1i02$pvalue,test_tw1i01$pvalue,test_tw1i05$pvalue,all_tw1i03$pvalue,all_tw1i02$pvalue,all_tw1i01$pvalue,all_tw1i05$pvalue)
                    


pvalues_tw1i_method<-c(rep("pvalues TCox (1-w) train set",4),rep("pvalues TCox (1-w) test set",4),rep("pvalues TCox (1-w) all set",4))




pvalues_tw1i<-data.frame(alpha_points,pvalues_tw1i_values,pvalues_tw1i_method)
p_pvalues_tw1i<-ggplot(data = pvalues_tw1i, aes(x = alpha_points, y = pvalues_tw1i_values,color= pvalues_tw1i_method)) + geom_line()+geom_point()+theme_minimal()
p_pvalues_tw1i+ ylim(0, max(pvalues_tw1i_values))
```



  
### exp((1-w)^3)

#### 0.3
```{r,warning=FALSE,message=FALSE}
#glmnet
cv.fit_twexp3 <- twiner_glmnet(xdata.train,ydata.train,0.3, tumornormal_weights4)


#genes
coefs.v_twexp3 <- twiner_coefs(cv.fit_twexp3)

fit.selected_twexp3 <- twiner_genes(cv.fit_twexp3)
length(fit.selected_twexp3)


#train
train_twexp03 <- train (coefs.v_twexp3)
train_twexp03$pvalue ## these values are the ones in Table 1
#test
test_twexp03 <- test (coefs.v_twexp3)
test_twexp03$pvalue
#all
all_twexp03 <- all (coefs.v_twexp3)

```

#### 0.2
```{r,warning=FALSE,message=FALSE}
#glmnet
cv.fit_twexp2 <- twiner_glmnet(xdata.train,ydata.train,0.2, tumornormal_weights4)


#genes
coefs.v_twexp2 <- twiner_coefs(cv.fit_twexp2)

fit.selected_twexp2 <- twiner_genes(cv.fit_twexp2)
length(fit.selected_twexp2)

#train
train_twexp02 <- train (coefs.v_twexp2)

#test
test_twexp02 <- test (coefs.v_twexp2)

#all
all_twexp02 <- all (coefs.v_twexp2)

```

#### 0.1
```{r,warning=FALSE,message=FALSE}
#glmnet
cv.fit_twexp1 <- twiner_glmnet(xdata.train,ydata.train,0.1, tumornormal_weights4)

#genes
coefs.v_twexp1 <- twiner_coefs(cv.fit_twexp1)

fit.selected_twexp1 <- twiner_genes(cv.fit_twexp1)
length(fit.selected_twexp1)

#train
train_twexp01 <- train (coefs.v_twexp1)

#test
test_twexp01 <- test (coefs.v_twexp1)

#all
all_twexp01 <- all (coefs.v_twexp1)

```

#### 0.05
```{r,warning=FALSE,message=FALSE}
#glmnet
cv.fit_twexp05 <- twiner_glmnet(xdata.train,ydata.train,0.05, tumornormal_weights4)

#genes
coefs.v_twexp05 <- twiner_coefs(cv.fit_twexp05)

fit.selected_twexp05 <- twiner_genes(cv.fit_twexp05)
length(fit.selected_twexp05)

#train
train_twexp05 <- train (coefs.v_twexp05)

#test
test_twexp05 <- test (coefs.v_twexp05)

#all
all_twexp05 <- all (coefs.v_twexp05)

```


#### plot

```{r,warning=FALSE,message=FALSE}
alpha_points<-c(0.3, 0.2,0.1,0.05)

pvalues_twexp_values_train<-c(train_twexp03$pvalue,train_twexp02$pvalue,train_twexp01$pvalue,train_twexp05$pvalue)

pvalues_twexp_values_test<-c(test_twexp03$pvalue,test_twexp02$pvalue,test_twexp01$pvalue,test_twexp05$pvalue)

pvalues_twexp_values_all<-c(all_twexp03$pvalue,all_twexp02$pvalue,all_twexp01$pvalue,all_twexp05$pvalue)



pvalues_twexp_values<-c(train_twexp03$pvalue,train_twexp02$pvalue,train_twexp01$pvalue,train_twexp05$pvalue,test_twexp03$pvalue,test_twexp02$pvalue,test_twexp01$pvalue,test_twexp05$pvalue,all_twexp03$pvalue,all_twexp02$pvalue,all_twexp01$pvalue,all_twexp05$pvalue)
                    


pvalues_twexp_method<-c(rep("pvalues TCox exp((1-w)^3) train set",4),rep("pvalues TCox exp((1-w)^3) test set",4),rep("pvalues TCox exp((1-w)^3) all set",4))




pvalues_twexp<-data.frame(alpha_points,pvalues_twexp_values,pvalues_twexp_method)
p_pvalues_twexp<-ggplot(data = pvalues_twexp, aes(x = alpha_points, y = pvalues_twexp_values,color= pvalues_twexp_method)) + geom_line()+geom_point()+theme_minimal()
p_pvalues_twexp+ ylim(0, max(pvalues_twexp_values))
```



### 1-normal weight^3

#### 0.3
```{r,warning=FALSE,message=FALSE}
#glmnet
cv.fit_twele3 <- twiner_glmnet(xdata.train,ydata.train,0.3, tumornormal_weights2)


#genes
coefs.v_twele3 <- twiner_coefs(cv.fit_twele3)

fit.selected_twele3 <- twiner_genes(cv.fit_twele3)
length(fit.selected_twele3)


#train
train_twele03 <- train (coefs.v_twele3)
train_twele03$pvalue ## these values are the ones in Table 1
#test
test_twele03 <- test (coefs.v_twele3)
test_twele03$pvalue
#all
all_twele03 <- all (coefs.v_twele3)

```

#### 0.2
```{r,warning=FALSE,message=FALSE}
#glmnet
cv.fit_twele2 <- twiner_glmnet(xdata.train,ydata.train,0.2, tumornormal_weights2)


#genes
coefs.v_twele2 <- twiner_coefs(cv.fit_twele2)

fit.selected_twele2 <- twiner_genes(cv.fit_twele2)
length(fit.selected_twele2)

#train
train_twele02 <- train (coefs.v_twele2)

#test
test_twele02 <- test (coefs.v_twele2)

#all
all_twele02 <- all (coefs.v_twele2)

```

#### 0.1
```{r,warning=FALSE,message=FALSE}
#glmnet
cv.fit_twele1 <- twiner_glmnet(xdata.train,ydata.train,0.1, tumornormal_weights2)

#genes
coefs.v_twele1 <- twiner_coefs(cv.fit_twele1)

fit.selected_twele1 <- twiner_genes(cv.fit_twele1)
length(fit.selected_twele1)

#train
train_twele01 <- train (coefs.v_twele1)

#test
test_twele01 <- test (coefs.v_twele1)

#all
all_twele01 <- all (coefs.v_twele1)

```

#### 0.05
```{r,warning=FALSE,message=FALSE}
#glmnet
cv.fit_twele05 <- twiner_glmnet(xdata.train,ydata.train,0.05, tumornormal_weights2)

#genes
coefs.v_twele05 <- twiner_coefs(cv.fit_twele05)

fit.selected_twele05 <- twiner_genes(cv.fit_twele05)
length(fit.selected_twele05)

#train
train_twele05 <- train (coefs.v_twele05)

#test
test_twele05 <- test (coefs.v_twele05)

#all
all_twele05 <- all (coefs.v_twele05)

```



#### plot

```{r,warning=FALSE,message=FALSE}
alpha_points<-c(0.3, 0.2,0.1,0.05)

pvalues_twele_values_train<-c(train_twele03$pvalue,train_twele02$pvalue,train_twele01$pvalue,train_twele05$pvalue)

pvalues_twele_values_test<-c(test_twele03$pvalue,test_twele02$pvalue,test_twele01$pvalue,test_twele05$pvalue)

pvalues_twele_values_all<-c(all_twele03$pvalue,all_twele02$pvalue,all_twele01$pvalue,all_twele05$pvalue)



pvalues_twele_values<-c(train_twele03$pvalue,train_twele02$pvalue,train_twele01$pvalue,train_twele05$pvalue,test_twele03$pvalue,test_twele02$pvalue,test_twele01$pvalue,test_twele05$pvalue,all_twele03$pvalue,all_twele02$pvalue,all_twele01$pvalue,all_twele05$pvalue)
                    


pvalues_twele_method<-c(rep("pvalues TCox (1-w^3) train set",4),rep("pvalues TCox (1-w^3) test set",4),rep("pvalues TCox (1-w^3) all set",4))




pvalues_twele<-data.frame(alpha_points,pvalues_twele_values,pvalues_twele_method)
p_pvalues_twele<-ggplot(data = pvalues_twele, aes(x = alpha_points, y = pvalues_twele_values,color= pvalues_twele_method)) + geom_line()+geom_point()+theme_minimal()
p_pvalues_twele+ ylim(0, max(pvalues_twele_values))
```



### (1-normal weight)^3
#### 0.3
```{r,warning=FALSE,message=FALSE}
#glmnet
cv.fit_tweleele3 <- twiner_glmnet(xdata.train,ydata.train,0.3, tumornormal_weights3)


#genes
coefs.v_tweleele3 <- twiner_coefs(cv.fit_tweleele3)

fit.selected_tweleele3 <- twiner_genes(cv.fit_tweleele3)
length(fit.selected_tweleele3)


#train
train_tweleele03 <- train (coefs.v_tweleele3)
train_tweleele03$pvalue ## these values are the ones in Table 1
#test
test_tweleele03 <- test (coefs.v_tweleele3)
test_tweleele03$pvalue
#all
all_tweleele03 <- all (coefs.v_tweleele3)

```
#### 0.2
```{r,warning=FALSE,message=FALSE}
#glmnet
cv.fit_tweleele2 <- twiner_glmnet(xdata.train,ydata.train,0.2, tumornormal_weights3)


#genes
coefs.v_tweleele2 <- twiner_coefs(cv.fit_tweleele2)

fit.selected_tweleele2 <- twiner_genes(cv.fit_tweleele2)
length(fit.selected_tweleele2)

#train
train_tweleele02 <- train (coefs.v_tweleele2)

#test
test_tweleele02 <- test (coefs.v_tweleele2)

#all
all_tweleele02 <- all (coefs.v_tweleele2)

```

#### 0.1
```{r,warning=FALSE,message=FALSE}
#glmnet
cv.fit_tweleele1 <- twiner_glmnet(xdata.train,ydata.train,0.1, tumornormal_weights3)

#genes
coefs.v_tweleele1 <- twiner_coefs(cv.fit_tweleele1)

fit.selected_tweleele1 <- twiner_genes(cv.fit_tweleele1)
length(fit.selected_tweleele1)

#train
train_tweleele01 <- train (coefs.v_tweleele1)

#test
test_tweleele01 <- test (coefs.v_tweleele1)

#all
all_tweleele01 <- all (coefs.v_tweleele1)

```

#### 0.05
```{r,warning=FALSE,message=FALSE}
#glmnet
cv.fit_tweleele05 <- twiner_glmnet(xdata.train,ydata.train,0.05, tumornormal_weights3)

#genes
coefs.v_tweleele05 <- twiner_coefs(cv.fit_tweleele05)

fit.selected_tweleele05 <- twiner_genes(cv.fit_tweleele05)
length(fit.selected_tweleele05)

#train
train_tweleele05 <- train (coefs.v_tweleele05)

#test
test_tweleele05 <- test (coefs.v_tweleele05)

#all
all_tweleele05 <- all (coefs.v_tweleele05)

```


#### plot

```{r,warning=FALSE,message=FALSE}
alpha_points<-c(0.3, 0.2,0.1,0.05)

pvalues_tweleele_values_train<-c(train_tweleele03$pvalue,train_tweleele02$pvalue,train_tweleele01$pvalue,train_tweleele05$pvalue)

pvalues_tweleele_values_test<-c(test_tweleele03$pvalue,test_tweleele02$pvalue,test_tweleele01$pvalue,test_tweleele05$pvalue)

pvalues_tweleele_values_all<-c(all_tweleele03$pvalue,all_tweleele02$pvalue,all_tweleele01$pvalue,all_tweleele05$pvalue)



pvalues_tweleele_values<-c(train_tweleele03$pvalue,train_tweleele02$pvalue,train_tweleele01$pvalue,train_tweleele05$pvalue,test_tweleele03$pvalue,test_tweleele02$pvalue,test_tweleele01$pvalue,test_tweleele05$pvalue,all_tweleele03$pvalue,all_tweleele02$pvalue,all_tweleele01$pvalue,all_tweleele05$pvalue)
                    


pvalues_tweleele_method<-c(rep("pvalues TCox (1-w)^3 train set",4),rep("pvalues TCox (1-w)^3 test set",4),rep("pvalues TCox (1-w)^3 all set",4))




pvalues_tweleele<-data.frame(alpha_points,pvalues_tweleele_values,pvalues_tweleele_method)
p_pvalues_tweleele<-ggplot(data = pvalues_tweleele, aes(x = alpha_points, y = pvalues_tweleele_values,color= pvalues_tweleele_method)) + geom_line()+geom_point()+theme_minimal()
p_pvalues_tweleele+ ylim(0, max(pvalues_tweleele_values))
```














### final plot with all weights - Figure 2

```{r,warning=FALSE,message=FALSE}
alpha_points<-c(0.3,0.2,0.1,0.05)

#1-w
pvalues_tw1i_values_test<-c(test_tw1i03$pvalue,test_tw1i02$pvalue,test_tw1i01$pvalue,test_tw1i05$pvalue)
#1/w
pvalues_twii_values_test<-c(test_twii03$pvalue,test_twii02$pvalue,test_twii01$pvalue,test_twii05$pvalue)
#exp((1-w)^3)
pvalues_twexp_values_test<-c(test_twexp03$pvalue,test_twexp02$pvalue,test_twexp01$pvalue,test_twexp05$pvalue)
#exp(-w^3)
pvalues_twexp2_values_test<-c(test_twexp203$pvalue,test_twexp202$pvalue,test_twexp201$pvalue,test_twexp205$pvalue)
#1-w^3
pvalues_twele_values_test<-c(test_twele03$pvalue,test_twele02$pvalue,test_twele01$pvalue,test_twele05$pvalue)
#(1-w)^3
pvalues_tweleele_values_test<-c(test_tweleele03$pvalue,test_tweleele02$pvalue,test_tweleele01$pvalue,test_tweleele05$pvalue)



pvalues_test<-c(pvalues_tw1i_values_test,pvalues_twii_values_test,pvalues_twexp_values_test,pvalues_twexp2_values_test,pvalues_twele_values_test,pvalues_tweleele_values_test)

pvalues_method_test<-c( rep("1-w",4), rep("1/w",4), rep("exp((1-w)^3)",4), rep("exp(-w^3)",4),rep("1-w^3 ",4),rep("(1-w)^3",4))

pvalues_test_data<-data.frame(alpha_points,pvalues_test,pvalues_method_test)
p_pvalues_test<-ggplot(data = pvalues_test_data, aes(x = alpha_points, y = pvalues_test,color= pvalues_method_test)) + geom_hline(yintercept=0.05) + geom_line()+geom_point()+theme_minimal()
p_pvalues_test+ ylim(0, max(pvalues_test))
```



## Demonstrative train and test survival curves - Figure 3 

Train and Test curves obtained using 1/w weight vector and alpha = 0.1

```{r,warning=FALSE,message=FALSE}
#train
train_twii01

#test
test_twii01
```


# Compare all models

## EN

### alpha=0.3

```{r,warning=FALSE,message=FALSE}
cv.fit_elastic_net3<-cv.glmnet(as.matrix(xdata.train),as.matrix(ydata.train), 
                               family = 'cox',foldid = my_foldid, 
                               alpha = 0.3,
                               nlambda = 100,
                               #lambda.min.ratio = 10^(-3),
                               nfolds= params$nfolds)

coefs.v_elastic3 <- coef(cv.fit_elastic_net3, s = 'lambda.min')[,1] %>% { .[. != 0]}

fit.selected_elastic3 <- which(cv.fit_elastic_net3$glmnet.fit$beta[,which(cv.fit_elastic_net3$cvm == min(cv.fit_elastic_net3$cvm))] != 0)

length(fit.selected_elastic3)
names(fit.selected_elastic3)
```


```{r,warning=FALSE,message=FALSE}
train_en03<-separate2GroupsCox(as.vector(coefs.v_elastic3), 
                               xdata.train[, names(coefs.v_elastic3)], 
                               ydata.train, 
                               plot.title = 'Train Set Elastic Net, alpha=0.3', legend.outside = FALSE)
train_en03$pvalue
test_en03<-separate2GroupsCox(as.vector(coefs.v_elastic3), 
                              xdata.test[, names(coefs.v_elastic3)], 
                              ydata.test, 
                              plot.title = 'Test Set Elastic Net, alpha=0.3', legend.outside = FALSE)
test_en03$pvalue
all_en03<-separate2GroupsCox(as.vector(coefs.v_elastic3), 
                             xdata[, names(coefs.v_elastic3)], 
                             ydata %>% mutate(time = time / 365 * 12), 
                             plot.title = 'All data Elastic Net, alpha=0.3', legend.outside = FALSE,
                             break.x.by = 12)
```

### alpha=0.2

```{r,warning=FALSE,message=FALSE}
cv.fit_elastic_net2<-cv.glmnet(as.matrix(xdata.train),as.matrix(ydata.train), 
                          family = 'cox',foldid = my_foldid, 
                          alpha = 0.2,
                        nlambda = 100,
               #lambda.min.ratio = 10^(-2),
                          nfolds= params$nfolds)

coefs.v_elastic2 <- coef(cv.fit_elastic_net2, s = 'lambda.min')[,1] %>% { .[. != 0]}

fit.selected_elastic2 <- which(cv.fit_elastic_net2$glmnet.fit$beta[,which(cv.fit_elastic_net2$cvm == min(cv.fit_elastic_net2$cvm))] != 0)

length(fit.selected_elastic2)
names(fit.selected_elastic2)
```


```{r,warning=FALSE,message=FALSE}
train_en02<-separate2GroupsCox(as.vector(coefs.v_elastic2), 
                   xdata.train[, names(coefs.v_elastic2)], 
                   ydata.train, 
                   plot.title = 'Train Set Elastic Net, alpha=0.2', legend.outside = FALSE)
train_en02$pvalue
test_en02<-separate2GroupsCox(as.vector(coefs.v_elastic2), 
                   xdata.test[, names(coefs.v_elastic2)], 
                   ydata.test, 
                   plot.title = 'Test Set Elastic Net, alpha=0.2', legend.outside = FALSE)
test_en02$pvalue
all_en02<-separate2GroupsCox(as.vector(coefs.v_elastic2), 
                   xdata[, names(coefs.v_elastic2)], 
                   ydata %>% mutate(time = time / 365 * 12), 
                   plot.title = 'All data Elastic Net, alpha=0.2', legend.outside = FALSE,
                   break.x.by = 12)
```

### alpha=0.1

```{r,warning=FALSE,message=FALSE}
cv.fit_elastic_net1<-cv.glmnet(as.matrix(xdata.train),as.matrix(ydata.train), 
                          family = 'cox',foldid = my_foldid, 
                          alpha = 0.1,
                        nlambda = 100,
               #lambda.min.ratio = 10^(-2),
                          nfolds= params$nfolds)


coefs.v_elastic1 <- coef(cv.fit_elastic_net1, s = 'lambda.min')[,1] %>% { .[. != 0]}

fit.selected_elastic1 <- which(cv.fit_elastic_net1$glmnet.fit$beta[,which(cv.fit_elastic_net1$cvm == min(cv.fit_elastic_net1$cvm))] != 0)

length(fit.selected_elastic1)
names(fit.selected_elastic1)
```

```{r,warning=FALSE,message=FALSE}
train_en01<-separate2GroupsCox(as.vector(coefs.v_elastic1), 
                   xdata.train[, names(coefs.v_elastic1)], 
                   ydata.train, 
                   plot.title = 'Train Set Elastic Net, alpha=0.1', legend.outside = FALSE)
train_en01$pvalue
test_en01<-separate2GroupsCox(as.vector(coefs.v_elastic1), 
                   xdata.test[, names(coefs.v_elastic1)], 
                   ydata.test, 
                   plot.title = 'Test Set Elastic Net, alpha=0.1', legend.outside = FALSE)
test_en01$pvalue
all_en01<-separate2GroupsCox(as.vector(coefs.v_elastic1), 
                   xdata[, names(coefs.v_elastic1)], 
                   ydata %>% mutate(time = time / 365 * 12), 
                   plot.title = 'All data Elastic Net, alpha=0.1', legend.outside = FALSE,
                   break.x.by = 12)
```

### alpha=0.05

```{r,warning=FALSE,message=FALSE}
cv.fit_elastic_net05<-cv.glmnet(as.matrix(xdata.train),as.matrix(ydata.train), 
                          family = 'cox',foldid = my_foldid, 
                          alpha = 0.05,
                        nlambda = 100,
               #lambda.min.ratio = 10^(-2),
                          nfolds= params$nfolds)

coefs.v_elastic05 <- coef(cv.fit_elastic_net05, s = 'lambda.min')[,1] %>% { .[. != 0]}

fit.selected_elastic05 <- which(cv.fit_elastic_net05$glmnet.fit$beta[,which(cv.fit_elastic_net05$cvm == min(cv.fit_elastic_net05$cvm))] != 0)

length(fit.selected_elastic05)
```

```{r,warning=FALSE,message=FALSE}
train_en05<-separate2GroupsCox(as.vector(coefs.v_elastic05), 
                   xdata.train[, names(coefs.v_elastic05)], 
                   ydata.train, 
                   plot.title = 'Train Set Elastic Net, alpha=0.05', legend.outside = FALSE)

test_en05<-separate2GroupsCox(as.vector(coefs.v_elastic05), 
                   xdata.test[, names(coefs.v_elastic05)], 
                   ydata.test, 
                   plot.title = 'Test Set Elastic Net, alpha=0.05', legend.outside = FALSE)

all_en05<-separate2GroupsCox(as.vector(coefs.v_elastic05), 
                   xdata[, names(coefs.v_elastic05)], 
                   ydata %>% mutate(time = time / 365 * 12), 
                   plot.title = 'All data Elastic Net, alpha=0.05', legend.outside = FALSE,
                   break.x.by = 12)
```


#####1 alpha=0.04

```{r,warning=FALSE,message=FALSE}
cv.fit_elastic_net04<-cv.glmnet(as.matrix(xdata.train),as.matrix(ydata.train), 
                          family = 'cox',foldid = my_foldid, 
                          alpha = 0.04,
                        nlambda = 100,
               #lambda.min.ratio = 10^(-2),
                          nfolds= params$nfolds)

coefs.v_elastic04 <- coef(cv.fit_elastic_net04, s = 'lambda.min')[,1] %>% { .[. != 0]}

fit.selected_elastic04 <- which(cv.fit_elastic_net04$glmnet.fit$beta[,which(cv.fit_elastic_net04$cvm == min(cv.fit_elastic_net04$cvm))] != 0)

length(fit.selected_elastic04)
```

```{r,warning=FALSE,message=FALSE}
train_en04<-separate2GroupsCox(as.vector(coefs.v_elastic04), 
                   xdata.train[, names(coefs.v_elastic04)], 
                   ydata.train, 
                   plot.title = 'Train Set Elastic Net, alpha=0.04', legend.outside = FALSE)

test_en04<-separate2GroupsCox(as.vector(coefs.v_elastic04), 
                   xdata.test[, names(coefs.v_elastic04)], 
                   ydata.test, 
                   plot.title = 'Test Set Elastic Net, alpha=0.04', legend.outside = FALSE)

all_en04<-separate2GroupsCox(as.vector(coefs.v_elastic04), 
                   xdata[, names(coefs.v_elastic04)], 
                   ydata %>% mutate(time = time / 365 * 12), 
                   plot.title = 'All data Elastic Net, alpha=0.04', legend.outside = FALSE,
                   break.x.by = 12)
```


### Pvalues KM

```{r,warning=FALSE,message=FALSE}
alpha_points<-c(0.3,0.2,0.1,0.05)

pvalues_en_train<-c(train_en03$pvalue,train_en02$pvalue,train_en01$pvalue,train_en05$pvalue)

pvalues_en_test<-c(test_en03$pvalue,test_en02$pvalue,test_en01$pvalue,test_en05$pvalue)

pvalues_en_all<-c(all_en03$pvalue,all_en02$pvalue,all_en01$pvalue,all_en05$pvalue)
```

```{r,warning=FALSE,message=FALSE}
pvalues_en_values<-c(train_en03$pvalue,train_en02$pvalue,train_en01$pvalue,train_en05$pvalue,
test_en03$pvalue,test_en02$pvalue,test_en01$pvalue,test_en05$pvalue,
all_en03$pvalue,all_en02$pvalue,all_en01$pvalue,all_en05$pvalue)

pvalues_en_method<-c(rep("pvalues en train set",4),rep("pvalues en test set",4),rep("pvalues en all set",4))
```

```{r,echo=FALSE}
pvalues_en<-data.frame(alpha_points,pvalues_en_values,pvalues_en_method)
p_pvalues_en<-ggplot(data = pvalues_en, aes(x = alpha_points, y = pvalues_en_values,color= pvalues_en_method)) + geom_line()+geom_point()+theme_minimal()
p_pvalues_en+ ylim(0, max(pvalues_en_values))
```




## Hub

### alpha=0.3

```{r,warning=FALSE,message=FALSE}
cv.fit_hub3<-cv.glmHub(as.matrix(xdata.train),as.matrix(ydata.train),
                       family = 'cox',foldid = my_foldid, network = "correlation",
                       alpha = 0.3,
                       nlambda = 100,
                       #lambda.min.ratio = 10^(-3),
                       nfolds= params$nfolds, network.options =  networkOptions(cutoff = 0.005,                                                                   min.degree = 0.6))
coefs.v_hub3 <- coef(cv.fit_hub3, s = 'lambda.min')[,1] %>% { .[. != 0]}

fit.selected_hub3 <- which(cv.fit_hub3$glmnet.fit$beta[,which(cv.fit_hub3$cvm == min(cv.fit_hub3$cvm))] != 0)

length(fit.selected_hub3)
names(fit.selected_hub3)
```


```{r,warning=FALSE,message=FALSE}
train_hub03<-separate2GroupsCox(as.vector(coefs.v_hub3),
                                xdata.train[, names(coefs.v_hub3)],
                                ydata.train,
                                plot.title = 'Train Set hub Net, alpha=0.3', legend.outside = FALSE)
train_hub03$pvalue

test_hub03<-separate2GroupsCox(as.vector(coefs.v_hub3),
                               xdata.test[, names(coefs.v_hub3)],
                               ydata.test,
                               plot.title = 'Test Set hub Net, alpha=0.3', legend.outside = FALSE)
test_hub03$pvalue

all_hub03<-separate2GroupsCox(as.vector(coefs.v_hub3),
                              xdata[, names(coefs.v_hub3)],
                              ydata %>% mutate(time = time / 365 * 12),
                              plot.title = 'All data hub Net, alpha=0.3', legend.outside = FALSE,
                              break.x.by = 12)
```





### alpha=0.2

```{r,warning=FALSE,message=FALSE}
cv.fit_hub2<-cv.glmHub(as.matrix(xdata.train),as.matrix(ydata.train),
                          family = 'cox',foldid = my_foldid, network = "correlation",
                          alpha = 0.2,
                        nlambda = 100,
               #lambda.min.ratio = 10^(-2),
                          nfolds= params$nfolds, network.options =  networkOptions(cutoff = 0.005,                                                                   min.degree = 0.6))
coefs.v_hub2 <- coef(cv.fit_hub2, s = 'lambda.min')[,1] %>% { .[. != 0]}

fit.selected_hub2 <- which(cv.fit_hub2$glmnet.fit$beta[,which(cv.fit_hub2$cvm == min(cv.fit_hub2$cvm))] != 0)

length(fit.selected_hub2)
names(fit.selected_hub2)
```


```{r,warning=FALSE,message=FALSE}
train_hub02<-separate2GroupsCox(as.vector(coefs.v_hub2),
                   xdata.train[, names(coefs.v_hub2)],
                   ydata.train,
                   plot.title = 'Train Set hub Net, alpha=0.2', legend.outside = FALSE)
train_hub02$pvalue

test_hub02<-separate2GroupsCox(as.vector(coefs.v_hub2),
                   xdata.test[, names(coefs.v_hub2)],
                   ydata.test,
                   plot.title = 'Test Set hub Net, alpha=0.2', legend.outside = FALSE)
test_hub02$pvalue

all_hub02<-separate2GroupsCox(as.vector(coefs.v_hub2),
                   xdata[, names(coefs.v_hub2)],
                   ydata %>% mutate(time = time / 365 * 12),
                   plot.title = 'All data hub Net, alpha=0.2', legend.outside = FALSE,
                   break.x.by = 12)
```


### alpha=0.1

```{r,warning=FALSE,message=FALSE}
cv.fit_hub1<-cv.glmHub(as.matrix(xdata.train),as.matrix(ydata.train), 
                          family = 'cox',foldid = my_foldid, network = "correlation", 
                          alpha = 0.1,
                        nlambda = 100,
               #lambda.min.ratio = 10^(-2),
                          nfolds= params$nfolds, network.options =  networkOptions(cutoff = 0.005,                                                                   min.degree = 0.6))


coefs.v_hub1 <- coef(cv.fit_hub1, s = 'lambda.min')[,1] %>% { .[. != 0]}

fit.selected_hub1 <- which(cv.fit_hub1$glmnet.fit$beta[,which(cv.fit_hub1$cvm == min(cv.fit_hub1$cvm))] != 0)

length(fit.selected_hub1)
names(fit.selected_hub1)
```


```{r,warning=FALSE,message=FALSE}
train_hub01<-separate2GroupsCox(as.vector(coefs.v_hub1), 
                   xdata.train[, names(coefs.v_hub1)], 
                   ydata.train, 
                   plot.title = 'Train Set hub Net, alpha=0.1', legend.outside = FALSE)
train_hub01$pvalue

test_hub01<-separate2GroupsCox(as.vector(coefs.v_hub1), 
                   xdata.test[, names(coefs.v_hub1)], 
                   ydata.test, 
                   plot.title = 'Test Set hub Net, alpha=0.1', legend.outside = FALSE)
test_hub01$pvalue

all_hub01<-separate2GroupsCox(as.vector(coefs.v_hub1), 
                   xdata[, names(coefs.v_hub1)], 
                   ydata %>% mutate(time = time / 365 * 12), 
                   plot.title = 'All data hub Net, alpha=0.1', legend.outside = FALSE,
                   break.x.by = 12)
```


### alpha=0.05

```{r,warning=FALSE,message=FALSE}
cv.fit_hub05<-cv.glmHub(as.matrix(xdata.train),as.matrix(ydata.train), 
                          family = 'cox',foldid = my_foldid, network = "correlation", 
                          alpha = 0.05,
                        nlambda = 100,
               #lambda.min.ratio = 10^(-2),
                          nfolds= params$nfolds, network.options =  networkOptions(cutoff = 0.005,                                                                   min.degree = 0.6))

coefs.v_hub05 <- coef(cv.fit_hub05, s = 'lambda.min')[,1] %>% { .[. != 0]}

fit.selected_hub05 <- which(cv.fit_hub05$glmnet.fit$beta[,which(cv.fit_hub05$cvm == min(cv.fit_hub05$cvm))] != 0)

length(fit.selected_hub05)
```

```{r,warning=FALSE,message=FALSE}
train_hub05<-separate2GroupsCox(as.vector(coefs.v_hub05), 
                   xdata.train[, names(coefs.v_hub05)], 
                   ydata.train, 
                   plot.title = 'Train Set hub Net, alpha=0.05', legend.outside = FALSE)

test_hub05<-separate2GroupsCox(as.vector(coefs.v_hub05), 
                   xdata.test[, names(coefs.v_hub05)], 
                   ydata.test, 
                   plot.title = 'Test Set hub Net, alpha=0.05', legend.outside = FALSE)

all_hub05<-separate2GroupsCox(as.vector(coefs.v_hub05), 
                   xdata[, names(coefs.v_hub05)], 
                   ydata %>% mutate(time = time / 365 * 12), 
                   plot.title = 'All data hub Net, alpha=0.05', legend.outside = FALSE,
                   break.x.by = 12)
```


### Pvalues KM

```{r,warning=FALSE,message=FALSE}
alpha_points<-c(0.3, 0.2,0.1,0.05)

pvalues_hub_train<-c(train_hub03$pvalue,train_hub02$pvalue,train_hub01$pvalue,train_hub05$pvalue)

pvalues_hub_test<-c(test_hub03$pvalue,test_hub02$pvalue,test_hub01$pvalue,test_hub05$pvalue)

pvalues_hub_all<-c(all_hub03$pvalue,all_hub02$pvalue,all_hub01$pvalue,all_hub05$pvalue)
```


```{r,warning=FALSE,message=FALSE}
pvalues_hub_values<-c(train_hub03$pvalue,train_hub02$pvalue,train_hub01$pvalue,train_hub05$pvalue,
                              test_hub03$pvalue,test_hub02$pvalue,test_hub01$pvalue,test_hub05$pvalue,
                              all_hub03$pvalue,all_hub02$pvalue,all_hub01$pvalue,all_hub05$pvalue)

pvalues_hub_method<-c(rep("pvalues hub train set",4),rep("pvalues hub test set",4),rep("pvalues hub all set",4))
 
pvalues_hub<-data.frame(alpha_points,pvalues_hub_values,pvalues_hub_method)
p_pvalues_hub<-ggplot(data = pvalues_hub, aes(x = alpha_points, y = pvalues_hub_values,color= pvalues_hub_method)) + geom_line()+geom_point()+theme_minimal()
p_pvalues_hub+ ylim(0, max(pvalues_hub_values))
```




## Orphan

### alpha=0.3

```{r,warning=FALSE,message=FALSE}
cv.fit_orphan3<-cv.glmOrphan(as.matrix(xdata.train),as.matrix(ydata.train), 
                             family = 'cox',foldid = my_foldid, network = "correlation", 
                             alpha = 0.3,
                             nlambda = 100,
                             #lambda.min.ratio = 10^(-3),
                             nfolds= params$nfolds, network.options =  networkOptions(cutoff = 0.001,                                                                   min.degree = 0.6))

coefs.v_orphan3 <- coef(cv.fit_orphan3, s = 'lambda.min')[,1] %>% { .[. != 0]}

fit.selected_orphan3 <- which(cv.fit_orphan3$glmnet.fit$beta[,which(cv.fit_orphan3$cvm == min(cv.fit_orphan3$cvm))] != 0)

length(fit.selected_orphan3)
names(fit.selected_orphan3)
```


```{r,warning=FALSE,message=FALSE}
train_orphan03<-separate2GroupsCox(as.vector(coefs.v_orphan3), 
                                   xdata.train[, names(coefs.v_orphan3)], 
                                   ydata.train, 
                                   plot.title = 'Train Set orphan Net, alpha=0.3', legend.outside = FALSE)
train_orphan03$pvalue

test_orphan03<-separate2GroupsCox(as.vector(coefs.v_orphan3), 
                                  xdata.test[, names(coefs.v_orphan3)], 
                                  ydata.test, 
                                  plot.title = 'Test Set orphan Net, alpha=0.3', legend.outside = FALSE)
test_orphan03$pvalue

all_orphan03<-separate2GroupsCox(as.vector(coefs.v_orphan3), 
                                 xdata[, names(coefs.v_orphan3)], 
                                 ydata %>% mutate(time = time / 365 * 12), 
                                 plot.title = 'All data orphan Net, alpha=0.3', legend.outside = FALSE,
                                 break.x.by = 12)
```
### alpha=0.2

```{r,warning=FALSE,message=FALSE}
cv.fit_orphan2<-cv.glmOrphan(as.matrix(xdata.train),as.matrix(ydata.train), 
                               family = 'cox',foldid = my_foldid, network = "correlation", 
                               alpha = 0.2,
                               nlambda = 100,
                               #lambda.min.ratio = 10^(-2),
                               nfolds= params$nfolds, network.options =  networkOptions(cutoff = 0.001,                                                                   min.degree = 0.6))

coefs.v_orphan2 <- coef(cv.fit_orphan2, s = 'lambda.min')[,1] %>% { .[. != 0]}

fit.selected_orphan2 <- which(cv.fit_orphan2$glmnet.fit$beta[,which(cv.fit_orphan2$cvm == min(cv.fit_orphan2$cvm))] != 0)

length(fit.selected_orphan2)
names(fit.selected_orphan2)
```


```{r,warning=FALSE,message=FALSE}
train_orphan02<-separate2GroupsCox(as.vector(coefs.v_orphan2), 
                               xdata.train[, names(coefs.v_orphan2)], 
                               ydata.train, 
                               plot.title = 'Train Set orphan Net, alpha=0.2', legend.outside = FALSE)
train_orphan02$pvalue

test_orphan02<-separate2GroupsCox(as.vector(coefs.v_orphan2), 
                              xdata.test[, names(coefs.v_orphan2)], 
                              ydata.test, 
                              plot.title = 'Test Set orphan Net, alpha=0.2', legend.outside = FALSE)
test_orphan02$pvalue

all_orphan02<-separate2GroupsCox(as.vector(coefs.v_orphan2), 
                             xdata[, names(coefs.v_orphan2)], 
                             ydata %>% mutate(time = time / 365 * 12), 
                             plot.title = 'All data orphan Net, alpha=0.2', legend.outside = FALSE,
                             break.x.by = 12)
```


### alpha=0.1

```{r,warning=FALSE,message=FALSE}
cv.fit_orphan1 <-cv.glmOrphan(as.matrix(xdata.train),as.matrix(ydata.train), 
                               family = 'cox',foldid = my_foldid, network = "correlation", 
                               alpha = 0.1,
                               nlambda = 100,
                               #lambda.min.ratio = 10^(-2),
                               nfolds= params$nfolds, network.options =  networkOptions(cutoff = 0.001,                                                                   min.degree = 0.6))

coefs.v_orphan1 <- coef(cv.fit_orphan1, s = 'lambda.min')[,1] %>% { .[. != 0]}

fit.selected_orphan1 <- which(cv.fit_orphan1$glmnet.fit$beta[,which(cv.fit_orphan1$cvm == min(cv.fit_orphan1$cvm))] != 0)

length(fit.selected_orphan1)
names(fit.selected_orphan1)
```


```{r,warning=FALSE,message=FALSE}
train_orphan01<-separate2GroupsCox(as.vector(coefs.v_orphan1), 
                               xdata.train[, names(coefs.v_orphan1)], 
                               ydata.train, 
                               plot.title = 'Train Set orphan Net, alpha=0.1', legend.outside = FALSE)
train_orphan01$pvalue

test_orphan01<-separate2GroupsCox(as.vector(coefs.v_orphan1), 
                              xdata.test[, names(coefs.v_orphan1)], 
                              ydata.test, 
                              plot.title = 'Test Set orphan Net, alpha=0.1', legend.outside = FALSE)
test_orphan01$pvalue
all_orphan01<-separate2GroupsCox(as.vector(coefs.v_orphan1), 
                             xdata[, names(coefs.v_orphan1)], 
                             ydata %>% mutate(time = time / 365 * 12), 
                             plot.title = 'All data orphan Net, alpha=0.1', legend.outside = FALSE,
                             break.x.by = 12)
```

### alpha=0.05

```{r,warning=FALSE,message=FALSE}
cv.fit_orphan05<-cv.glmOrphan(as.matrix(xdata.train),as.matrix(ydata.train), 
                               family = 'cox',foldid = my_foldid, network = "correlation", 
                               alpha = 0.05,
                               nlambda = 100,
                               #lambda.min.ratio = 10^(-2),
                               nfolds= params$nfolds, network.options =  networkOptions(cutoff = 0.001,                                                                   min.degree = 0.6))

coefs.v_orphan05 <- coef(cv.fit_orphan05, s = 'lambda.min')[,1] %>% { .[. != 0]}

fit.selected_orphan05 <- which(cv.fit_orphan05$glmnet.fit$beta[,which(cv.fit_orphan05$cvm == min(cv.fit_orphan05$cvm))] != 0)

length(fit.selected_orphan05)
```


```{r,warning=FALSE,message=FALSE}
train_orphan05<-separate2GroupsCox(as.vector(coefs.v_orphan05), 
                               xdata.train[, names(coefs.v_orphan05)], 
                               ydata.train, 
                               plot.title = 'Train Set orphan Net, alpha=0.05', legend.outside = FALSE)

test_orphan05<-separate2GroupsCox(as.vector(coefs.v_orphan05), 
                              xdata.test[, names(coefs.v_orphan05)], 
                              ydata.test, 
                              plot.title = 'Test Set orphan Net, alpha=0.05', legend.outside = FALSE)

all_orphan05<-separate2GroupsCox(as.vector(coefs.v_orphan05), 
                             xdata[, names(coefs.v_orphan05)], 
                             ydata %>% mutate(time = time / 365 * 12), 
                             plot.title = 'All data orphan Net, alpha=0.05', legend.outside = FALSE,
                             break.x.by = 12)
```



### Pvalues KM


```{r,warning=FALSE,message=FALSE}
alpha_points<-c(0.3,0.2,0.1,0.05)

pvalues_orphan_train<-c(train_orphan03$pvalue,train_orphan02$pvalue,train_orphan01$pvalue,train_orphan05$pvalue)

pvalues_orphan_test<-c(test_orphan03$pvalue,test_orphan02$pvalue,test_orphan01$pvalue,test_orphan05$pvalue)

pvalues_orphan_all<-c(all_orphan03$pvalue,all_orphan02$pvalue,all_orphan01$pvalue,all_orphan05$pvalue)

```

```{r,warning=FALSE,message=FALSE}
pvalues_orphan_values<-c(train_orphan03$pvalue,train_orphan02$pvalue,train_orphan01$pvalue,train_orphan05$pvalue,
                      test_orphan03$pvalue,test_orphan02$pvalue,test_orphan01$pvalue,test_orphan05$pvalue,
                      all_orphan03$pvalue,all_orphan02$pvalue,all_orphan01$pvalue,all_orphan05$pvalue)

pvalues_orphan_method<-c(rep("pvalues orphan train set",4),rep("pvalues orphan test set",4),rep("pvalues orphan all set",4))


pvalues_orphan<-data.frame(alpha_points,pvalues_orphan_values,pvalues_orphan_method)
p_pvalues_orphan<-ggplot(data = pvalues_orphan, aes(x = alpha_points, y = pvalues_orphan_values,color= pvalues_orphan_method)) + geom_line()+geom_point()+theme_minimal()
p_pvalues_orphan+ ylim(0, max(pvalues_orphan_values))
```
















## Final plot comparing all models - Figure 4
```{r,warning=FALSE,message=FALSE}
alpha_points<-c(0.3, 0.2,0.1,0.05)

#TCox (exp(-w^3))
pvalues_twexp2_values_test<-c(test_twii03$pvalue,test_twii02$pvalue,test_twii01$pvalue,test_twii05$pvalue)

#EN
pvalues_en_test<-c(test_en03$pvalue,test_en02$pvalue,test_en01$pvalue,test_en05$pvalue)

#Hub
pvalues_hub_test<-c(test_hub03$pvalue,test_hub02$pvalue,test_hub01$pvalue,test_hub05$pvalue)

#Orphan
pvalues_orphan_test<-c(test_orphan03$pvalue,test_orphan02$pvalue,test_orphan01$pvalue,test_orphan05$pvalue)

pvalues_test<-c(pvalues_twexp2_values_test, pvalues_en_test, pvalues_hub_test, pvalues_orphan_test)

Method <-c(rep("TCox",4), rep("EN",4), rep("HubCox",4), rep("OrphanCox",4))

pvalues_test_data<-data.frame(alpha_points,pvalues_test,Method)

p_pvalues_test<-ggplot(data = pvalues_test_data, aes(x = alpha_points, y = pvalues_test,color= Method)) + geom_hline(yintercept=0.05) + geom_line()+geom_point()+theme_minimal()
p_pvalues_test+ ylim(0, max(pvalues_test))
```

```{r,warning=FALSE,message=FALSE}
alpha_points<-c(0.3, 0.2,0.1,0.05)

#TCox (exp(-w^3))
pvalues_twexp2_values_train<-c(train_twexp203$pvalue,train_twexp202$pvalue,train_twexp201$pvalue,train_twexp205$pvalue)

#EN
pvalues_en_train<-c(train_en03$pvalue,train_en02$pvalue,train_en01$pvalue,train_en05$pvalue)

#Hub
pvalues_hub_train<-c(train_hub03$pvalue,train_hub02$pvalue,train_hub01$pvalue,train_hub05$pvalue)

#Orphan
pvalues_orphan_train<-c(train_orphan03$pvalue,train_orphan02$pvalue,train_orphan01$pvalue,train_orphan05$pvalue)

pvalues_train<-c(pvalues_twexp2_values_train, pvalues_en_train, pvalues_hub_train, pvalues_orphan_train)

Method <-c(rep("TCox",4), rep("EN",4), rep("HubCox",4), rep("OrphanCox",4))

pvalues_train_data<-data.frame(alpha_points,pvalues_train,Method)

p_pvalues_train<-ggplot(data = pvalues_train_data, aes(x = alpha_points, y = pvalues_train,color= Method)) + geom_hline(yintercept=0.05) + geom_line()+geom_point()+theme_minimal()
p_pvalues_train+ ylim(0, max(pvalues_train))
```


# CHAT tool

by all models
```{r,warning=FALSE,message=FALSE}
genes <- tumorf[,c("CYP7A1","FAM159A","LOC732275","CLDN9","LBX2","MEIG1","PAX5","NKAIN4","ZDHHC19","GRAPL","PCDHB12","EEPD1","HPCAL1","PGAM2","ZNF883","FAM138B","LOC646498","PRCD")]

geneNames(names(genes)) %>% { hallmarks(.$external_gene_name)$heatmap} + theme(title = element_text(size=9),axis.text=element_text(size=10), axis.title=element_text(size=8))


```

genes selected only by:

EN

```{r,warning=FALSE,message=FALSE}
genes <- tumorf[,c("HOTAIR","GJA3","LOC283663","DNAI2","NELF","GUCA1B")]

geneNames(names(genes)) %>% { hallmarks(.$external_gene_name)$heatmap} + theme(title = element_text(size=9),axis.text=element_text(size=10), axis.title=element_text(size=8))
```

glmSparseNet

```{r,warning=FALSE,message=FALSE}
genes <- tumorf[,c("CYGB","UNC13B","LIPT2","RFT1","BEND4","FAM24B","RASGRP2","SLFN11")]

geneNames(names(genes)) %>% { hallmarks(.$external_gene_name)$heatmap} + theme(title = element_text(size=9),axis.text=element_text(size=10), axis.title=element_text(size=8))
```


TCox
```{r,warning=FALSE,message=FALSE}
genes <- tumorf[,c("ANKRD26P1","CARKD","IGLON5","OSTN","RAB20","TXNL4B","AOX2P","DCLK3","FCRL2","SEPT7P2","ASPHD1","COL19A1","DCP1A","FLJ16779","LOC100303728","PCDHA7","SNTG1","COX4I2","NXF2B","TAC3","C20orf106","LOC285780","OR2T5","TERF2IP","CAPN7","OSBPL3","TRIM67")]

geneNames(names(genes)) %>% { hallmarks(.$external_gene_name)$heatmap} + theme(title = element_text(size=9),axis.text=element_text(size=10), axis.title=element_text(size=8))
```



# Differentially expressed genes

## Early vs. advanced stage

```{r,warning=FALSE,message=FALSE}
clinic$neoplasm_diseasestage[clinic$neoplasm_diseasestage == "stage ivb"] <- 4
clinic$neoplasm_diseasestage[clinic$neoplasm_diseasestage == "stage iva"] <- 4
clinic$neoplasm_diseasestage[clinic$neoplasm_diseasestage == "stage iv"] <- 4
clinic$neoplasm_diseasestage[clinic$neoplasm_diseasestage == "stage iiic"] <- 3
clinic$neoplasm_diseasestage[clinic$neoplasm_diseasestage == "stage iiib"] <- 3
clinic$neoplasm_diseasestage[clinic$neoplasm_diseasestage == "stage iiia"] <- 3
clinic$neoplasm_diseasestage[clinic$neoplasm_diseasestage == "stage iii"] <- 3
clinic$neoplasm_diseasestage[clinic$neoplasm_diseasestage == "stage iic"] <- 2
clinic$neoplasm_diseasestage[clinic$neoplasm_diseasestage == "stage iib"] <- 2
clinic$neoplasm_diseasestage[clinic$neoplasm_diseasestage == "stage iia"] <- 2
clinic$neoplasm_diseasestage[clinic$neoplasm_diseasestage == "stage ii"] <- 2
clinic$neoplasm_diseasestage[clinic$neoplasm_diseasestage == "stage ia"] <- 1
clinic$neoplasm_diseasestage[clinic$neoplasm_diseasestage == "stage i"] <- 1

pathology_stage <- as.data.frame(clinic$neoplasm_diseasestage)
pathology_stage$names <- rownames(clinic)
pathology_stage$names <- toupper(pathology_stage$names)  
pathology_stage$names <- chartr('.', '-', pathology_stage$names)
pathology_stage <- na.omit(pathology_stage)
row.names(pathology_stage) <- pathology_stage$names
pathology_stage <- as.data.frame(pathology_stage)
pathology_stage$`clinic$neoplasm_diseasestage` <- as.numeric(pathology_stage$`clinic$neoplasm_diseasestage`)

pathology_stage$`clinic$neoplasm_diseasestage`[pathology_stage$`clinic$neoplasm_diseasestage` == "1"] <- "early"
pathology_stage$`clinic$neoplasm_diseasestage`[pathology_stage$`clinic$neoplasm_diseasestage` == "2"] <-  "early"
pathology_stage$`clinic$neoplasm_diseasestage`[pathology_stage$`clinic$neoplasm_diseasestage` == "3"] <-  "early"
pathology_stage$`clinic$neoplasm_diseasestage`[pathology_stage$`clinic$neoplasm_diseasestage` == "4"] <- "advanced"



all_data = merge(tumorf,pathology_stage,by="row.names")
#rownames(all_data) <- all_data$submitter_id

xdata <- apply(as.matrix(all_data[,2:20502]),2,as.numeric)
xdata <- xdata[ ,- as.numeric(which(apply(xdata, 2, var) == 0))]
xdata <- t(xdata)

group <- all_data$`clinic$neoplasm_diseasestage`


```

```{r,warning=FALSE,message=FALSE}
y <- DGEList(counts=xdata,group=group)
#remove genes that are unexpressed or very lowly expressed in the samples
keep <- rowSums(cpm(y)>1) >= 2
y <- y[keep, , keep.lib.sizes=FALSE]

y <- calcNormFactors(y)
# plotMDS(y)

design <- model.matrix(~group)
y <- estimateDisp(y,design)
plotBCV(y)
# y$samples
# levels(y$samples$group)

```

To perform quasi-likelihood F-tests
(The quasi-likelihood method is highly
recommended for differential expression analyses of bulk RNA-seq data as it gives stricter
error rate control by accounting for the uncertainty in dispersion estimation)

```{r,warning=FALSE,message=FALSE}
fit <- glmQLFit(y, design)

qlf <- glmQLFTest(fit,coef=2)


# lrt <- glmLRT(fit)
topTags(qlf)

summary(decideTests(qlf))

a <- qlf$table
a <- a[a$PValue < 0.05, ] 

write.table(a, "tabela_genes_degs_earlyvsadv.csv", append = TRUE)

```
 
Plot log-fold change against log-counts per million, with DE genes highlighted:

```{r,warning=FALSE,message=FALSE}
plotMD(qlf)
abline(h=c(-1, 1), col="blue")
```



## Tumor vs. normal 

```{r,warning=FALSE,message=FALSE}
# tumor 373
# normal 60
tumorf <- t(tumorf)
normalf <- t(normalf)
all = merge(tumorf,normalf,by="row.names")
rownames(all) <- all$Row.names

all <- all[,-1]
all <- t(all)
all <- as.data.frame(all)
all$type[1:433] <- "normal"
all$type[1:373] <- "tumor"


xdatadeg <- apply(as.matrix(all[,1:20501]),2,as.numeric)
xdatadeg <- xdatadeg[,apply(xdatadeg!=0, 2, all)]
xdatadeg <- t(xdatadeg)

group <- all$type

```

```{r,warning=FALSE,message=FALSE}
y <- DGEList(counts=xdatadeg,group=group)
#remove genes that are unexpressed or very lowly expressed in the samples
keep <- rowSums(cpm(y)>1) >= 2
y <- y[keep, , keep.lib.sizes=FALSE]

y <- calcNormFactors(y)
# plotMDS(y)

design <- model.matrix(~group)
y <- estimateDisp(y,design)
plotBCV(y)
# y$samples
# levels(y$samples$group)

```

To perform quasi-likelihood F-tests
(The quasi-likelihood method is highly
recommended for differential expression analyses of bulk RNA-seq data as it gives stricter
error rate control by accounting for the uncertainty in dispersion estimation)

```{r,warning=FALSE,message=FALSE}
fit <- glmQLFit(y, design)

qlf <- glmQLFTest(fit,coef=2)


# lrt <- glmLRT(fit)
topTags(qlf)

summary(decideTests(qlf))

b <- qlf$table
b <- b[a$PValue < 0.05, ] 

write.table(b, "tabela_genes_degs_tumorvsnorm.csv", append = TRUE)

```
 
Plot log-fold change against log-counts per million, with DE genes highlighted:

```{r,warning=FALSE,message=FALSE}
plotMD(qlf)
abline(h=c(-1, 1), col="blue")
```


## Tumor vs. normal genes selected by all models

```{r,warning=FALSE,message=FALSE}
# tumor 373
# normal 60

few <- all[,c("CYP7A1",
"FAM159A",
"LOC732275",
"CLDN9",
"LBX2",
"MEIG1",
"PAX5",
"NKAIN4",
"ZDHHC19",
"GRAPL",
"PCDHB12",
"EEPD1",
"HPCAL1",
"PGAM2",
"ZNF883",
"FAM138B",
"LOC646498",
"PRCD","type")]

xdatadeg <- apply(as.matrix(few[,1:18]),2,as.numeric)
xdatadeg <- t(xdatadeg)

group <- all$type

```

```{r,warning=FALSE,message=FALSE}
y <- DGEList(counts=xdatadeg,group=group)
#remove genes that are unexpressed or very lowly expressed in the samples
keep <- rowSums(cpm(y)>1) >= 2
y <- y[keep, , keep.lib.sizes=FALSE]

y <- calcNormFactors(y)
# plotMDS(y)

design <- model.matrix(~group)
y <- estimateDisp(y,design)
plotBCV(y)
# y$samples
# levels(y$samples$group)

```

```{r,warning=FALSE,message=FALSE}
fit <- glmQLFit(y, design)

qlf <- glmQLFTest(fit,coef=2)


# lrt <- glmLRT(fit)
topTags(qlf)

summary(decideTests(qlf))
decideTests(qlf)

```

```{r,warning=FALSE,message=FALSE}
plotMD(qlf)
abline(h=c(-1, 1), col="blue")
```

## Tumor vs. normal genes selected by TCox

```{r,warning=FALSE,message=FALSE}
# tumor 373
# normal 60

few <- all[,c("ANKRD26P1",
"CARKD",
"IGLON5",
"OSTN",
"RAB20",
"TXNL4B",
"AOX2P",
"DCLK3",
"FCRL2",
"SEPT7P2",
"ASPHD1",
"COL19A1",
"DCP1A",
"FLJ16779",
"LOC100303728",
"PCDHA7",
"SNTG1",
"COX4I2",
"NXF2B",
"TAC3",
"C20orf106",
"LOC285780",
"OR2T5",
"TERF2IP",
"CAPN7",
"OSBPL3",
"TRIM67",
"type")]

xdatadeg <- apply(as.matrix(few[,1:27]),2,as.numeric)
xdatadeg <- t(xdatadeg)

group <- all$type

```

```{r,warning=FALSE,message=FALSE}
y <- DGEList(counts=xdatadeg,group=group)
#remove genes that are unexpressed or very lowly expressed in the samples
keep <- rowSums(cpm(y)>1) >= 2
y <- y[keep, , keep.lib.sizes=FALSE]

y <- calcNormFactors(y)
# plotMDS(y)

design <- model.matrix(~group)
y <- estimateDisp(y,design)
plotBCV(y)
# y$samples
# levels(y$samples$group)

```

```{r,warning=FALSE,message=FALSE}
fit <- glmQLFit(y, design)

qlf <- glmQLFTest(fit,coef=2)


# lrt <- glmLRT(fit)
topTags(qlf)

summary(decideTests(qlf))
decideTests(qlf)

qlf$table

```

```{r,warning=FALSE,message=FALSE}
plotMD(qlf)
abline(h=c(-1, 1), col="blue")
```
## Tumor vs. normal genes selected by EN

```{r,warning=FALSE,message=FALSE}
# tumor 373
# normal 60

few <- all[,c("HOTAIR", "GJA3", "LOC283663", "DNAI2", "NELF", "GUCA1B",
"type")]

xdatadeg <- apply(as.matrix(few[,1:6]),2,as.numeric)
xdatadeg <- t(xdatadeg)

group <- all$type

```

```{r,warning=FALSE,message=FALSE}
y <- DGEList(counts=xdatadeg,group=group)
#remove genes that are unexpressed or very lowly expressed in the samples
keep <- rowSums(cpm(y)>1) >= 2
y <- y[keep, , keep.lib.sizes=FALSE]

y <- calcNormFactors(y)
# plotMDS(y)

design <- model.matrix(~group)
y <- estimateDisp(y,design)
plotBCV(y)
# y$samples
# levels(y$samples$group)

```

```{r,warning=FALSE,message=FALSE}
fit <- glmQLFit(y, design)

qlf <- glmQLFTest(fit,coef=2)


# lrt <- glmLRT(fit)
topTags(qlf)

summary(decideTests(qlf))
decideTests(qlf)

qlf$table

```

```{r,warning=FALSE,message=FALSE}
plotMD(qlf)
abline(h=c(-1, 1), col="blue")
```
## Tumor vs. normal genes selected by HUB

```{r,warning=FALSE,message=FALSE}
# tumor 373
# normal 60

few <- all[,c("CYGB", "UNC13B", "LIPT2", "RFT1", "BEND4", "FAM24B", "RASGRP2", "SLFN11",
"type")]

xdatadeg <- apply(as.matrix(few[,1:8]),2,as.numeric)
xdatadeg <- t(xdatadeg)

group <- all$type

```

```{r,warning=FALSE,message=FALSE}
y <- DGEList(counts=xdatadeg,group=group)
#remove genes that are unexpressed or very lowly expressed in the samples
keep <- rowSums(cpm(y)>1) >= 2
y <- y[keep, , keep.lib.sizes=FALSE]

y <- calcNormFactors(y)
# plotMDS(y)

design <- model.matrix(~group)
y <- estimateDisp(y,design)
plotBCV(y)
# y$samples
# levels(y$samples$group)

```

```{r,warning=FALSE,message=FALSE}
fit <- glmQLFit(y, design)

qlf <- glmQLFTest(fit,coef=2)


# lrt <- glmLRT(fit)
topTags(qlf)

summary(decideTests(qlf))
decideTests(qlf)

qlf$table

```

```{r,warning=FALSE,message=FALSE}
plotMD(qlf)
abline(h=c(-1, 1), col="blue")
```
## Tumor vs. normal genes selected by all models + TCox + EN + HUB

```{r,warning=FALSE,message=FALSE}
# tumor 373
# normal 60

few <- all[,c("CYP7A1",
"FAM159A",
"LOC732275",
"CLDN9",
"LBX2",
"MEIG1",
"PAX5",
"NKAIN4",
"ZDHHC19",
"GRAPL",
"PCDHB12",
"EEPD1",
"HPCAL1",
"PGAM2",
"ZNF883",
"FAM138B",
"LOC646498",
"PRCD",
"ANKRD26P1",
"CARKD",
"IGLON5",
"OSTN",
"RAB20",
"TXNL4B",
"AOX2P",
"DCLK3",
"FCRL2",
"SEPT7P2",
"ASPHD1",
"COL19A1",
"DCP1A",
"FLJ16779",
"LOC100303728",
"PCDHA7",
"SNTG1",
"COX4I2",
"NXF2B",
"TAC3",
"C20orf106",
"LOC285780",
"OR2T5",
"TERF2IP",
"CAPN7",
"OSBPL3",
"TRIM67","HOTAIR", "GJA3", "LOC283663", "DNAI2", "NELF", "GUCA1B", "CYGB", "UNC13B", "LIPT2", "RFT1", "BEND4", "FAM24B", "RASGRP2", "SLFN11",
"type")]

xdatadeg <- apply(as.matrix(few[,1:59]),2,as.numeric)
xdatadeg <- t(xdatadeg)

group <- all$type

```

```{r,warning=FALSE,message=FALSE}
y <- DGEList(counts=xdatadeg,group=group)
#remove genes that are unexpressed or very lowly expressed in the samples
keep <- rowSums(cpm(y)>1) >= 2
y <- y[keep, , keep.lib.sizes=FALSE]

y <- calcNormFactors(y)
# plotMDS(y)

design <- model.matrix(~group)
y <- estimateDisp(y,design)
plotBCV(y)
# y$samples
# levels(y$samples$group)

```

```{r,warning=FALSE,message=FALSE}
fit <- glmQLFit(y, design)

qlf <- glmQLFTest(fit,coef=2)


# lrt <- glmLRT(fit)
topTags(qlf)

summary(decideTests(qlf))
decideTests(qlf)

qlf$table

```

```{r,warning=FALSE,message=FALSE}
plotMD(qlf)
abline(h=c(-1, 1), col="blue")
``` 
# Survival with genes selected


ANKRD26P1
```{r}
tumorf <- as.data.frame(t(tumorf))
gene <- as.data.frame(tumorf$ANKRD26P1)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$ANKRD26P1` <= median(gene$`tumorf$ANKRD26P1`)] <- "low"
gene$type[gene$`tumorf$ANKRD26P1` > median(gene$`tumorf$ANKRD26P1`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE, title = "ANKRD26P1")



# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```


CARKD
```{r}
gene <- as.data.frame(tumorf$CARKD)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$CARKD` <= median(gene$`tumorf$CARKD`)] <- "low"
gene$type[gene$`tumorf$CARKD` > median(gene$`tumorf$CARKD`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "CARKD")



# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```


IGLON5
```{r}
gene <- as.data.frame(tumorf$IGLON5)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$IGLON5` <= median(gene$`tumorf$IGLON5`)] <- "low"
gene$type[gene$`tumorf$IGLON5` > median(gene$`tumorf$IGLON5`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "IGLON5")



# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```

OSTN
```{r}
gene <- as.data.frame(tumorf$OSTN)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$OSTN` <= median(gene$`tumorf$OSTN`)] <- "low"
gene$type[gene$`tumorf$OSTN` > median(gene$`tumorf$OSTN`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "OSTN")



# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```



RAB20
```{r}
gene <- as.data.frame(tumorf$RAB20)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$RAB20` <= median(gene$`tumorf$RAB20`)] <- "low"
gene$type[gene$`tumorf$RAB20` > median(gene$`tumorf$RAB20`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "RAB20")




# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```


TXNL4B
```{r}
gene <- as.data.frame(tumorf$TXNL4B)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$TXNL4B` <= median(gene$`tumorf$TXNL4B`)] <- "low"
gene$type[gene$`tumorf$TXNL4B` > median(gene$`tumorf$TXNL4B`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "TXNL4B")




# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```


AOX2P
```{r}
gene <- as.data.frame(tumorf$AOX2P)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$AOX2P` <= median(gene$`tumorf$AOX2P`)] <- "low"
gene$type[gene$`tumorf$AOX2P` > median(gene$`tumorf$AOX2P`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "AOX2P")



# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```

DCLK3
```{r}
gene <- as.data.frame(tumorf$DCLK3)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$DCLK3` <= median(gene$`tumorf$DCLK3`)] <- "low"
gene$type[gene$`tumorf$DCLK3` > median(gene$`tumorf$DCLK3`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "DCLK3")




# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```


FCRL2
```{r}
gene <- as.data.frame(tumorf$FCRL2)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$FCRL2` <= median(gene$`tumorf$FCRL2`)] <- "low"
gene$type[gene$`tumorf$FCRL2` > median(gene$`tumorf$FCRL2`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)
ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "FCRL2")




# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```


SEPT7P2
```{r}
gene <- as.data.frame(tumorf$SEPT7P2)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$SEPT7P2` <= median(gene$`tumorf$SEPT7P2`)] <- "low"
gene$type[gene$`tumorf$SEPT7P2` > median(gene$`tumorf$SEPT7P2`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "SEPT7P2")




# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```


ASPHD1
```{r}
gene <- as.data.frame(tumorf$ASPHD1)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$ASPHD1` <= median(gene$`tumorf$ASPHD1`)] <- "low"
gene$type[gene$`tumorf$ASPHD1` > median(gene$`tumorf$ASPHD1`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "ASPHD1")




# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```


COL19A1
```{r}
gene <- as.data.frame(tumorf$COL19A1)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$COL19A1` <= median(gene$`tumorf$COL19A1`)] <- "low"
gene$type[gene$`tumorf$COL19A1` > median(gene$`tumorf$COL19A1`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "COL19A1")




# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```


DCP1A
```{r}
gene <- as.data.frame(tumorf$DCP1A)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$DCP1A` <= median(gene$`tumorf$DCP1A`)] <- "low"
gene$type[gene$`tumorf$DCP1A` > median(gene$`tumorf$DCP1A`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "DCP1A")



# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```


FLJ16779
```{r}
gene <- as.data.frame(tumorf$FLJ16779)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$FLJ16779` <= median(gene$`tumorf$FLJ16779`)] <- "low"
gene$type[gene$`tumorf$FLJ16779` > median(gene$`tumorf$FLJ16779`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "FLJ16779")




# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```

LOC100303728
```{r}
gene <- as.data.frame(tumorf$LOC100303728)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$LOC100303728` <= median(gene$`tumorf$LOC100303728`)] <- "low"
gene$type[gene$`tumorf$LOC100303728` > median(gene$`tumorf$LOC100303728`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "LOC100303728")




# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```


PCDHA7
```{r}
gene <- as.data.frame(tumorf$PCDHA7)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$PCDHA7` <= median(gene$`tumorf$PCDHA7`)] <- "low"
gene$type[gene$`tumorf$PCDHA7` > median(gene$`tumorf$PCDHA7`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "PCDHA7")



# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```

SNTG1
```{r}
gene <- as.data.frame(tumorf$SNTG1)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$SNTG1` <= median(gene$`tumorf$SNTG1`)] <- "low"
gene$type[gene$`tumorf$SNTG1` > median(gene$`tumorf$SNTG1`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "SNTG1")




# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```


COX4I2
```{r}
gene <- as.data.frame(tumorf$COX4I2)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$COX4I2` <= median(gene$`tumorf$COX4I2`)] <- "low"
gene$type[gene$`tumorf$COX4I2` > median(gene$`tumorf$COX4I2`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "COX4I2")




# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```

NXF2B
```{r}
gene <- as.data.frame(tumorf$NXF2B)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$NXF2B` <= median(gene$`tumorf$NXF2B`)] <- "low"
gene$type[gene$`tumorf$NXF2B` > median(gene$`tumorf$NXF2B`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)
ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "NXF2B")




# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```

TAC3
```{r}
gene <- as.data.frame(tumorf$TAC3)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$TAC3` <= median(gene$`tumorf$TAC3`)] <- "low"
gene$type[gene$`tumorf$TAC3` > median(gene$`tumorf$TAC3`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "TAC3")




# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```

C20orf106
```{r}
gene <- as.data.frame(tumorf$C20orf106)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$C20orf106` <= median(gene$`tumorf$C20orf106`)] <- "low"
gene$type[gene$`tumorf$C20orf106` > median(gene$`tumorf$C20orf106`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "C20orf106")




# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```

LOC285780
```{r}
gene <- as.data.frame(tumorf$LOC285780)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$LOC285780` <= median(gene$`tumorf$LOC285780`)] <- "low"
gene$type[gene$`tumorf$LOC285780` > median(gene$`tumorf$LOC285780`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "LOC285780")




# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```


OR2T5
```{r}
gene <- as.data.frame(tumorf$OR2T5)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$OR2T5` <= median(gene$`tumorf$OR2T5`)] <- "low"
gene$type[gene$`tumorf$OR2T5` > median(gene$`tumorf$OR2T5`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "OR2T5")




# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```


TERF2IP
```{r}
gene <- as.data.frame(tumorf$TERF2IP)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$TERF2IP` <= median(gene$`tumorf$TERF2IP`)] <- "low"
gene$type[gene$`tumorf$TERF2IP` > median(gene$`tumorf$TERF2IP`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)



plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "TERF2IP")




# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```

CAPN7
```{r}
gene <- as.data.frame(tumorf$CAPN7)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$CAPN7` <= median(gene$`tumorf$CAPN7`)] <- "low"
gene$type[gene$`tumorf$CAPN7` > median(gene$`tumorf$CAPN7`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "CAPN7")




# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```

OSBPL3
```{r}
gene <- as.data.frame(tumorf$OSBPL3)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$OSBPL3` <= median(gene$`tumorf$OSBPL3`)] <- "low"
gene$type[gene$`tumorf$OSBPL3` > median(gene$`tumorf$OSBPL3`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "OSBPL3")





# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```

TRIM67
```{r}
gene <- as.data.frame(tumorf$TRIM67)
gene$rownames <- rownames(tumorf)

gene$type[gene$`tumorf$TRIM67` <= median(gene$`tumorf$TRIM67`)] <- "low"
gene$type[gene$`tumorf$TRIM67` > median(gene$`tumorf$TRIM67`)] <- "high"

rownames(gene) <- gene$rownames

pathology_stage <- merge(datasurvf, gene, by="row.names")

time <- as.numeric(pathology_stage$time)
status <- pathology_stage$status
path_stage <- pathology_stage$type


# Surv object
kirc.surv = Surv(time,status == 1)
#kirc.surv 
# +did not died


# Kaplan-Meier estimates
fitkir <- survfit(Surv(as.numeric(time,status)) ~ path_stage)
#fitkir
#summary(fitkir)

plot(fitkir)

ggsurvplot(fitkir, data = pathology_stage, pval = TRUE,title = "TRIM671")




# the log/rank test
surv.stage <- survdiff(Surv(time,status==1)~path_stage)
surv.stage

```

